application/src/main/java/io/smallrye/context/application/context/propagation/ApplicationContextProvider.java

```diff
@@ -8,6 +10,28 @@
 
 public class ApplicationContextProvider implements ThreadContextProvider {
 
+    static final ClassLoader SYSTEM_CL;
+
+    static {
+        final SecurityManager sm = System.getSecurityManager();
+        if (sm != null) {
+            SYSTEM_CL = AccessController
+                    .doPrivileged((PrivilegedAction<ClassLoader>) ApplicationContextProvider::calculateSystemClassLoader);
```

Comment:
@FroMage If we need privileged action to grab the CL, don't we also need it to set it (inside `currentContext` and `clearContext()`)?

Reply:
I just copied this from SR-Config, I didn't give it much thought.

Reply:
Got link to that SR config code?

Reply:
https://github.com/smallrye/smallrye-config/blob/master/implementation/src/main/java/io/smallrye/config/SmallRyeConfigProviderResolver.java#L47

Reply:
Ok, let's keep it this way for now. We can always fix it up later.